server/command.go

```diff
@@ -0,0 +1,136 @@
+		return "Please specify an action for /zoom command.", nil
+	}
+
+	userID := args.UserId
+	user, appErr := p.API.GetUser(userID)
+	if appErr != nil {
+		p.postCommandResponse(args, fmt.Sprintf("We could not retrieve user (userId: %v)", args.UserId))
+		return fmt.Sprintf("We could not retrieve user (userId: %v)", args.UserId), nil
+	}
+
+	if action == "start" {
+		if _, appErr = p.API.GetChannelMember(args.ChannelId, userID); appErr != nil {
+			p.postCommandResponse(args, fmt.Sprintf("We could not get channel members (channelId: %v)", args.ChannelId))
+			return fmt.Sprintf("We could not get channel members (channelId: %v)", args.ChannelId), nil
+		}
+
+		meetingID := 0
+		personal := false
+
+		// Determine if the user is sending command in DM or channel
+		channel, tmpErr := p.API.GetChannel(args.ChannelId)
+		if tmpErr != nil {
+			p.postCommandResponse(args, fmt.Sprintf("We could not get channel members (channelId: %v)", args.ChannelId))
+			return fmt.Sprintf("We could not get channel members (channelId: %v)", args.ChannelId), nil
+		}
+
+		if channel.Type == "D" {
+			// create a personal zoom meeting
+			personal = true
+			ru, clientErr := p.zoomClient.GetUser(user.Email)
+			if clientErr != nil {
+				p.postCommandResponse(args, "We could not verify your Mattermost account in Zoom. Please ensure that your Mattermost email address matches your Zoom login email address.")
+				return "We could not verify your Mattermost account in Zoom. Please ensure that your Mattermost email address matches your Zoom login email address.", nil
+			}
+			meetingID = ru.Pmi
+		} else {
+			// create a channel zoom meeting
+			meeting := &zoom.Meeting{
+				Type: zoom.MeetingTypeInstant,
+			}
+
+			rm, clientErr := p.zoomClient.CreateMeeting(meeting, user.Email)
+			if clientErr != nil {
+				p.postCommandResponse(args, "We could not create and start a meeting in Zoom. Please ensure that your Mattermost email address matches your Zoom login email address.")
+				return "We could not create and start a meeting in Zoom. Please ensure that your Mattermost email address matches your Zoom login email address.", nil
+			}
+			meetingID = rm.ID
+		}
+
+		zoomURL := strings.TrimSpace(config.ZoomURL)
+		if len(zoomURL) == 0 {
+			zoomURL = "https://zoom.us"
+		}
+
+		meetingURL := fmt.Sprintf("%s/j/%v", zoomURL, meetingID)
+
+		post := &model.Post{
+			UserId:    p.botUserID,
+			ChannelId: args.ChannelId,
+			Message:   fmt.Sprintf("Meeting started at %s.", meetingURL),
+			Type:      "custom_zoom",
+			Props: map[string]interface{}{
+				"meeting_id":       meetingID,
+				"meeting_link":     meetingURL,
+				"meeting_status":   zoom.WebhookStatusStarted,
+				"meeting_personal": personal,
+			},
+		}
+
+		_, appErr := p.API.CreatePost(post)
+		if appErr != nil {
+			p.postCommandResponse(args, "Failed to post message. Please try again.")
+			return "Failed to post message. Please try again.", nil
+		}
+	}
+	return "", nil
```

Comment:
Can you please post an error in this case?
```suggestion
	return fmt.Sprintf("Unknown action %v", action) , nil
```

Reply:
@dhadiseputro What do you think about this?

Reply:
This part of code is executed in both cases (successful or not). I can add `unknown action` but by adding an `else` case. What do you think?

Reply:
Or you can ` return "", nil` at the end if the if statement. Thats up to you.